.github/workflows/analysis_workflow.yml

```diff
@@ -262,14 +265,14 @@ jobs:
           python -m asv continuous -v --show-stderr origin/master HEAD -f 1.15
 
       - name: Publish results
-        if: github.event_name != 'pull_request' || github.ref == 'refs/heads/master'
+        # synchronize: when a pull request is merged
+        if: inputs.force_publish_result == true || github.event_name == 'schedule' || github.event_name == 'synchronize'
```

Comment:
Can you remind me why we want `github.event_name == 'synchronize'? What is this trying to achieve?

Reply:
When the PR is merged. It's to replace `github.ref == 'refs/heads/master'`. The old one, will trigger the publish of result if a user manually triggers a run on the master branch. 
In my PR, variable `force_publish_result` is added so user can have more fine-grained control of whether simply running the test with master branch, or run and publish.